pkg/agent/controller/noderoute/node_route_controller.go

```diff
@@ -381,13 +409,12 @@ func (c *Controller) syncNodeRoute(nodeName string) error {
 func (c *Controller) deleteNodeRoute(nodeName string) error {
 	klog.Infof("Deleting routes and flows to Node %s", nodeName)
 
-	route, flowsAreInstalled := c.installedNodes.Load(nodeName)
-	if route != nil {
-		if err := netlink.RouteDel(route.(*netlink.Route)); err != nil {
-			return fmt.Errorf("failed to delete the route to Node %s: %v", nodeName, err)
-		}
-		c.installedNodes.Store(nodeName, nil)
+	entry, flowsAreInstalled := c.installedNodes.Load(nodeName)
+	routes := entry.([]*netlink.Route)
```

Comment:
Need to check `routes != nil` like before, otherwise the conversion would panic if the node is not installed before.

Reply:
r u concerned about iterate through nil? go can iterate through nil slice, it produced 0 iterations. but compiles and runs.

func main() {
    var test []string 
    for _, t := range test {
        fmt.Printf("no harm, %s", t)
    }
    fmt.Print("done")
}

Reply:
No, my concern is the conversion, nil (type interface{}) can not be converted to the slice. Try this: https://play.golang.org/p/AXHOGXPZvay

Reply:
done.